core/service/options.go

```diff
@@ -1,33 +1,45 @@
 package service
 
 import (
+	"github.com/libp2p/go-libp2p-core/crypto"
 	"github.com/textileio/go-threads/core/thread"
 	"github.com/textileio/go-threads/crypto/symmetric"
 )
 
-// AddOptions defines options for adding a new thread.
-type AddOptions struct {
+// KeyOptions defines options for keys when creating / adding a thread.
+type KeyOptions struct {
 	FollowKey *symmetric.Key
 	ReadKey   *symmetric.Key
+	LogKey    crypto.Key
```

Comment:
This allows a client to dictate the log private key. Can be either private or public.

Reply:
This is great @sanderpick. In addition to addressing this 'issue', it opens us up to a whole slew of possibilities re: identity sources, which is awesome. Essentially, anything that satisfies this interface will do, which is pretty awesome, good idea!